sentry-android-core/src/main/java/io/sentry/android/core/SystemEventsBreadcrumbsIntegration.java

```diff
@@ -92,46 +115,168 @@ public void register(final @NotNull IScopes scopes, final @NotNull SentryOptions
+
+    if (!options.isEnableSystemEventBreadcrumbs()) {
+      return;
+    }
+
+    try (final @NotNull ISentryLifecycleToken ignored = receiverLock.acquire()) {
+      if (isClosed || isStopped || receiver != null) {
+        return;
+      }
+    }
+
+    try {
+      options
+          .getExecutorService()
+          .submit(
+              () -> {
+                try (final @NotNull ISentryLifecycleToken ignored = receiverLock.acquire()) {
+                  if (isClosed || isStopped || receiver != null) {
+                    return;
+                  }
+
+                  receiver = new SystemEventsBroadcastReceiver(scopes, options);
+                  if (filter == null) {
+                    filter = new IntentFilter();
+                    for (String item : actions) {
+                      filter.addAction(item);
                     }
                   }
-                });
-      } catch (Throwable e) {
-        options
-            .getLogger()
-            .log(
-                SentryLevel.DEBUG,
-                "Failed to start SystemEventsBreadcrumbsIntegration on executor thread.",
-                e);
-      }
+                  try {
+                    // registerReceiver can throw SecurityException but it's not documented in the
+                    // official docs
+                    ContextUtils.registerReceiver(context, options, receiver, filter);
+                    if (newIntegration) {
+                      options
+                          .getLogger()
+                          .log(SentryLevel.DEBUG, "SystemEventsBreadcrumbsIntegration installed.");
+                      addIntegrationToSdkVersion("SystemEventsBreadcrumbs");
+                    }
+                  } catch (Throwable e) {
+                    options.setEnableSystemEventBreadcrumbs(false);
+                    options
+                        .getLogger()
+                        .log(
+                            SentryLevel.ERROR,
+                            "Failed to initialize SystemEventsBreadcrumbsIntegration.",
+                            e);
+                  }
+                }
+              });
+    } catch (Throwable e) {
+      options
+          .getLogger()
+          .log(
+              SentryLevel.WARNING,
+              "Failed to start SystemEventsBreadcrumbsIntegration on executor thread.");
     }
   }
 
-  private void startSystemEventsReceiver(
-      final @NotNull IScopes scopes, final @NotNull SentryAndroidOptions options) {
-    receiver = new SystemEventsBroadcastReceiver(scopes, options);
-    final IntentFilter filter = new IntentFilter();
-    for (String item : actions) {
-      filter.addAction(item);
+  private void unregisterReceiver() {
+    try (final @NotNull ISentryLifecycleToken ignored = receiverLock.acquire()) {
+      isStopped = true;
     }
+
+    if (receiver != null) {
+      context.unregisterReceiver(receiver);
+      receiver = null;
+    }
```

Comment:
I'd also put writing receiver inside the lock block, so it aligns with `registerReceiver()`
```suggestion
    try (final @NotNull ISentryLifecycleToken ignored = receiverLock.acquire()) {
      isStopped = true;
      if (receiver != null) {
        context.unregisterReceiver(receiver);
        receiver = null;
      }
    }
```

Reply:
I guess the idea here was to only lock over the boolean flag and don't block for the heavy `unregisterReceiver` call. Since we check the for the boolean flag in `registerReceiver` (inside the same lock), it should be fine? Or do you see a case where this would still be racy?

Reply:
Oh I see, yeah I would still keep the write to receiver within the block. How about the following?

```java
final @Nullable SystemEventsBroadcastReceiver receiverRef = null;

try (final @NotNull ISentryLifecycleToken ignored = receiverLock.acquire()) {
  isStopped = true;
  receiverRef = receiver;
  receiver = null;
}

if (receiverRef != null) {
  context.unregisterReceiver(receiverRef);
}
```

Reply:
that makes sense, thanks!